<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ClusteringRingsWindow</class>
 <widget class="QMainWindow" name="ClusteringRingsWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>444</width>
    <height>444</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Clustering Rings</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>401</width>
      <height>391</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="generation_tab">
     <attribute name="title">
      <string>Generation</string>
     </attribute>
     <widget class="QLineEdit" name="num_instances_input">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>20</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Number of intances to generate per type of instance (clean, extends, collides)</string>
      </property>
      <property name="toolTipDuration">
       <number>-1</number>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>10</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="num_circs_input">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>60</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Number of circumferences per instance</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>3</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="randomness_input">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>100</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Randomness in the x and y axis when creating points of a circurference (value proportional to radius)</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>1.0</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="range_points_input">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>140</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>From this range a random number will be selected to choose the number of points of a circumference</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>100, 150</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="range_radius_input">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>180</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>From this range a random value will be chose for each circumference radius</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>5.0, 15.0</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>20</y>
        <width>191</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Number Of Instances</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>60</y>
        <width>171</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Number Of Circumferences</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>100</y>
        <width>181</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Randomness</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>140</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Range of points</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>180</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Radius value range</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>220</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Noise Ratio</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="noise_ratio_input">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>220</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Ammount of noise points per circumference point (The higher, the less accurate the clustering algorithm)</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>0.05</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="dataset_input_generation">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>260</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Directory in which the instances will be saved</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>./dataset</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>260</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Output path</string>
      </property>
     </widget>
     <widget class="QPushButton" name="generate_button">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>310</y>
        <width>85</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="detection_tab">
     <attribute name="title">
      <string>Detection</string>
     </attribute>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>40</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Dataset path</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="dataset_input_detection">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>40</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Directory where the instances are located</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>./dataset</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="fuzziness_input">
      <property name="geometry">
       <rect>
        <x>251</x>
        <y>80</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Range from which a random fuzziness value will be chosen per attempt to calculate the clusters</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>1.05, 2.0</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>80</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Fuzziness range</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>120</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Attempts per instance</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="attempts_input">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>120</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Number of times we run the algorithm per instance</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>40</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>160</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Max Iterations</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="max_iter_input">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>160</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>In case of no convergence, maximum number of iterations before halting</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>100</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>200</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Membership thresshold</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="mem_thress_input">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>200</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Minimum membership value for a point to be considered part of a cluster when calculating the radius</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>0.3</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>240</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Results Output path</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="results_input_detection">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>240</y>
        <width>113</width>
        <height>29</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Directory where the results will be saved</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>./results</string>
      </property>
     </widget>
     <widget class="QPushButton" name="detect_button">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>300</y>
        <width>101</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Run Detection</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="visuals_tab">
     <attribute name="title">
      <string>Visuals</string>
     </attribute>
     <widget class="QFrame" name="frame">
      <property name="geometry">
       <rect>
        <x>2</x>
        <y>1</y>
        <width>393</width>
        <height>141</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>30</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Dataset path</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="dataset_input_visuals">
       <property name="geometry">
        <rect>
         <x>212</x>
         <y>30</y>
         <width>151</width>
         <height>29</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>Directory where the instances are located</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="placeholderText">
        <string>./dataset</string>
       </property>
      </widget>
      <widget class="QPushButton" name="vis_dataset_button">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>90</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Visualize instances</string>
       </property>
      </widget>
     </widget>
     <widget class="QFrame" name="frame_3">
      <property name="geometry">
       <rect>
        <x>2</x>
        <y>144</y>
        <width>393</width>
        <height>212</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QLabel" name="label_16">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>40</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Results Json path</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="results_input_visuals">
       <property name="geometry">
        <rect>
         <x>212</x>
         <y>40</y>
         <width>151</width>
         <height>29</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>Directory where the results are located</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="placeholderText">
        <string>./results/results_2023-04-27 13:47:46.254407.json</string>
       </property>
      </widget>
      <widget class="QPushButton" name="vis_results_button">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>90</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Visualize Detection</string>
       </property>
      </widget>
      <widget class="QPushButton" name="stats_results_button">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>150</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Visualize Statistics</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
